File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        Deque<Integer> q = new ArrayDeque<>();
        
        q.offerLast(arr[0]);
        
        for (int i = 1; i < arr.length; i++) {
            int now = q.peekLast();
            if (now != arr[i]) {
                q.offerLast(arr[i]);
            }
        }
        
        int[] result = new int[q.size()];
        
        for (int i = 0; i < result.length; i++) {
            result[i] = q.pollFirst();
        }
        
        return result;
    }
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
def solution(arr):
    answer = []
    
    # 배열을 순회하면서 각 요소와 비교
    for i in arr: # 원소
        # 현재 answer가 비어있거나, answer의 마지막 값과 현재 arr의 원소가 같을때
        if len(answer) == 0 or answer[-1] != i:
            # answer에 집어넣기
            answer.append(i)
    
    return answer